packages/client/src/search/Search.spec.ts

```diff
@@ -17,6 +17,22 @@ describe(`Given the ${Search.name} configured to work with sandbox`, () => {
         }),
       ).resolves.not.toThrow();
     });
+
+    it('should paginate to the next page if there are more results', async () => {
+      const search = new Search(testConfig);
+
+      const result = await search.searchProfiles({
+        query: 'test',
+        limit: 10,
+      });
+
+      expect(result.items.length).not.toBe(0);
+
+      const nextResult = await result.next();
```

Comment:
nit: an open question given we are modeling here the public api:

Can we assume that no one would try to save a reference to the first `result` and try to always call `next` from there? 
Below is a potential pseudo code:

```
const list = await search.searchProfiles({
        query: 'test',
        limit: 10,
      }); 

const secondPage = await list.next();
const thirdPage =  await list.next(); // noop, still second page really
```